myLib/httpH/httpH.go

```diff
@@ -0,0 +1,616 @@
+
+	row := sqlDB.QueryRow("SELECT id, date, title, comment, repeat FROM scheduler WHERE id = :id",
+		sql.Named("id", id))
+
+	err = row.Scan(&msg.ID, &msg.Date, &msg.Title, &msg.Comment, &msg.Repeat)
+
+	if err != nil {
+		w.Write(dataSendError)
+		return
+	}
+
+	// Анализ содержимого в поле date после проверки
+	// Если поле пустое - происходит удаление задачи по принятому параметру id
+	// Если поле не пустое - сохраняются новые данные по принятому параметру id
+	//
+	if msg.Repeat == "" { // Удаление записи
+
+		_, err = sqlDB.Exec("DELETE FROM scheduler WHERE id = :id",
+			sql.Named("id", id))
+
+		if err != nil {
+			w.Write(dataSendError)
+			return
+		}
+
+	} else { // Обновление записи
+
+		// Проверка поля repeat
+		//
+		tn := time.Now()
+
+		if err != nil {
+			w.Write(dataSendError)
+			return
+		}
+
+		msg.Date, err = genDate.NextDate(tn, msg.Date, msg.Repeat)
+		if err != nil {
+			w.Write(dataSendError)
+			return
+		}
+
+		_, err = sqlDB.Exec("UPDATE scheduler SET date = :date, title = :title, comment = :comment, repeat = :repeat WHERE id = :id",
+			sql.Named("date", msg.Date),
+			sql.Named("title", msg.Title),
+			sql.Named("comment", msg.Comment),
+			sql.Named("repeat", msg.Repeat),
+			sql.Named("id", id))
+
+		if err != nil {
+			w.Write(dataSendError)
+			return
+		}
+
+	}
+
+	w.WriteHeader(http.StatusOK)
+	w.Write([]byte("{}"))
+
+}
+
+// Удаление задачи
+func DelTasksH(w http.ResponseWriter, r *http.Request) {
+
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+
+	// Подготовка аварийного сообщения в случае ошибки обработчика
+	//
+	dataSendError, err := generationArrMsg("Задача не найдена")
+	if err != nil {
+		http.Error(w, `"Internal Server Error"`, http.StatusInternalServerError)
+		return
+	}
+
+	// Чтение параметра запроса
+	//
+	idQ := r.URL.Query().Get("id")
+
+	id, err := qualId(idQ)
+	if err != nil {
+		w.Write(dataSendError)
+		return
+	}
+
+	// Подключение к БД
+	//
+	sqlDB, err := dbA.CheckCreateDB()
+	if err != nil {
+		log.Fatal(err)
+		os.Exit(1)
+	}
+	defer func() {
+		err = sqlDB.Close()
+		if err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	// Удаление задачи из БД
+	//
+	_, err = sqlDB.Exec("DELETE FROM scheduler WHERE id = :id",
+		sql.Named("id", id))
+
+	if err != nil {
+		w.Write(dataSendError)
+		return
+	}
+
+	// Успешное завершение
+	//
+	w.WriteHeader(http.StatusOK)
+	w.Write([]byte("{}"))
+
+}
+
+// Функция возвращает массив байт и ошибку,
+//
+// Параметры:
+//
+// str - строка по данным которой формируется массив байт
+func generationArrMsg(str string) (b []byte, err error) {
+
+	errMsg := udt.ErrMsgFormat{
+		Error: str,
+	}
+	b, err = json.Marshal(errMsg)
+	if err != nil {
+		return nil, err
+	}
+
+	return b, nil
+}
+
+// Функция производит проверку содержимого id. Возвращает признак id формате int и ошибку
+//
+// Параметры:
+//
+// id - строка с номером
+func qualId(idQ string) (int, error) {
+
+	if idQ == "" {
+		return 0, errors.New("Ошибка")
```

Comment:
Опять же не понятна суть ошибки
```suggestion
		return 0, errors.New("Empty string")
```